python_modules/libraries/dagster-components/dagster_components/lib/dbt_project/component.py

```diff
@@ -10,48 +10,56 @@
     DbtProject,
     dbt_assets,
 )
-from pydantic import BaseModel
-from typing_extensions import Self
 
 from dagster_components import Component, ComponentLoadContext
 from dagster_components.core.component import component_type
+from dagster_components.core.schema.base import ResolvableModel
 from dagster_components.core.schema.metadata import ResolvableFieldInfo
 from dagster_components.core.schema.objects import (
     AssetAttributesModel,
     AssetSpecTransformModel,
-    OpSpecBaseModel,
+    OpSpecModel,
     TemplatedValueResolver,
 )
 from dagster_components.lib.dbt_project.scaffolder import DbtProjectComponentScaffolder
 from dagster_components.utils import ResolvingInfo, get_wrapped_translator_class
 
 
-class DbtProjectParams(BaseModel):
+class DbtProjectParams(ResolvableModel["DbtProjectComponent"]):
     dbt: DbtCliResource
-    op: Optional[OpSpecBaseModel] = None
+    op: Optional[OpSpecModel] = None
     asset_attributes: Annotated[
         Optional[AssetAttributesModel], ResolvableFieldInfo(required_scope={"node"})
     ] = None
     transforms: Optional[Sequence[AssetSpecTransformModel]] = None
 
+    def resolve(self, resolver: TemplatedValueResolver) -> "DbtProjectComponent":
+        return DbtProjectComponent(
+            resource=self.dbt,
+            op_spec=self.op.resolve(resolver) if self.op else None,
+            translator=get_wrapped_translator_class(DagsterDbtTranslator)(
+                resolving_info=ResolvingInfo(
+                    "node", self.asset_attributes or AssetAttributesModel(), resolver
+                )
+            ),
+            transforms=[transform.resolve(resolver) for transform in (self.transforms or [])],
+        )
+
 
 @component_type(name="dbt_project")
 class DbtProjectComponent(Component):
     def __init__(
         self,
-        dbt_resource: DbtCliResource,
-        op_spec: Optional[OpSpecBaseModel],
-        asset_attributes: Optional[AssetAttributesModel],
-        transforms: Sequence[AssetSpecTransformModel],
-        value_resolver: TemplatedValueResolver,
+        resource: DbtCliResource,
+        op_spec: Optional[OpSpecModel],
+        translator: DagsterDbtTranslator,
+        transforms: Sequence[Callable[[Definitions], Definitions]],
```

Comment:
this is fairly strange type signature. what is going on here?

Reply:
why isn't this just `Sequence[Definitions]` and can we call it `defs_sequence` instead of transforms?

Reply:
@schrockn these are the python-land version of:

```yaml
transforms:
  target: "group:a"
  attributes:
    owners: ["owen@dagsterlabs.com"]
```

I thought the most natural / general representation of that was a function that takes in a Definitions object and returns a modified one, otherwise we'd need to create some sort of specialized object to do this (or just not resolve the underlying transform objects)